main/main.cpp

```diff
@@ -2761,6 +2761,16 @@ Error Main::setup2(bool p_show_boot_logo) {
 		OS::get_singleton()->benchmark_end_measure("Servers", "Modules and Extensions");
 	}
 
+	// Perform this check before initializing the rest of the servers to avoid a crash on failure.
+	if (Engine::get_singleton()->get_write_movie_path() != String()) {
+		movie_writer = MovieWriter::find_writer_for_file(Engine::get_singleton()->get_write_movie_path());
+		if (movie_writer == nullptr) {
+			ERR_PRINT(vformat("Can't find MovieWriter for file type, aborting.\nCheck if the specified file path has a valid extension: %s", Engine::get_singleton()->get_write_movie_path()));
+			OS::get_singleton()->set_exit_code(EXIT_FAILURE);
+			return FAILED;
```

Comment:
For the clean exit, it's likely needs the same cleanup as DisplayServer failure:

https://github.com/godotengine/godot/blob/913a0c0043daa3009737ffa45c09744631167b80/main/main.cpp#L2857-L2879